metrics/web/test/common/presentation/navigation/state/navigation_notifier_test.dart

```diff
@@ -167,13 +168,91 @@ void main() {
       },
     );
 
+    test(
+      ".handleLoggedOut() clears pages when a user logs out",
+      () {
+        notifier.handleLoggedIn();
+        notifier.push(DefaultRoutes.dashboard);
+
+        notifier.handleLoggedOut();
+
+        expect(notifier.pages, hasLength(1));
+      },
+    );
+
+    test(
+      ".handleLoggedOut() pushes to the login page when a user logs out",
+      () {
+        notifier.handleLoggedIn();
+        notifier.push(DefaultRoutes.dashboard);
+
+        notifier.handleLoggedOut();
+
+        final currentPage = notifier.pages.last;
+
+        expect(currentPage.name, isLoginPageName);
+      },
+    );
+
+    test(
+      ".handleLoggedOut() does not clear pages when a user logs out from the login page",
+      () {
+        notifier.push(DefaultRoutes.login);
+
+        notifier.handleLoggedOut();
+
+        expect(notifier.pages, hasLength(2));
+      },
+    );
+
+    test(
+      ".handleLoggedOut() does not push to the login page when a user logs out from the login page",
+      () {
+        notifier.handleLoggedOut();
+
+        final loginPagesCount = notifier.pages
+            .where((page) => page.routeName == RouteName.login)
+            .length;
+
+        expect(loginPagesCount, equals(1));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() pushes the dashboard route if the current redirect route is null",
+      () {
+        notifier.handleLoggedIn();
+
+        final currentPage = notifier.pages.last;
+
+        expect(currentPage.name, isDashboardPageName);
+      },
+    );
+
+    test(
+      ".handleLoggedIn() pushes the redirect route if it is not null",
+      () {
+        final expectedRouteConfiguration = DefaultRoutes.projectGroups;
+
+        notifier.handleAppInitialized(isAppInitialized: false);
+        notifier.handleInitialRoutePath(expectedRouteConfiguration);
+        notifier.handleAppInitialized(isAppInitialized: true);
+
+        notifier.handleLoggedIn();
+
+        final currentPage = notifier.pages.last;
+
+        expect(currentPage.routeName, expectedRouteConfiguration.name);
+      },
+    );
+
     test(
       ".handleAuthenticationUpdates() clears pages and pushes to the login page when the user logs out",
```

Comment:
✅ 
It seems like there is no `handleAuthenticationUpdates` method in the `NavigationNotifier` class.